### PR TITLE
[3.10] Refactor glusterfs for scaleup + upgrade

### PR DESCRIPTION
**Pull request by jarrpa**
These commits refactor and enhance the GlusterFS playbooks and roles to work with OpenShift cluster scaleups and provide automation for GlusterFS upgrades.

Cherry-pick from: https://github.com/openshift/openshift-ansible/pull/9365 https://github.com/openshift/openshift-ansible/pull/9468

### CODE REVIEW

**Comment by jarrpa** [comment]
./retest

File: roles/openshift_storage_glusterfs/tasks/glusterfs_common.yml [hunk 1]
```diff
@@ -315,12 +101,10 @@
...
 
-- include_tasks: glusterblock_deploy.yml
-  when:
-  - glusterfs_block_deploy
-  #TODO: Remove this when multipathd will be available on atomic
-  - not openshift_is_atomic | bool
+- import_tasks: glusterblock_deploy.yml
+  when: glusterfs_block_deploy
 
+<<<<<<< HEAD
```
Comment (michaelgugino) [inline roles/openshift_storage_glusterfs/tasks/glusterfs_common.yml]: @jarrpa bad cherry-pick here.

Need to reconcile.

**Comment by jarrpa** [comment]
@michaelgugino Oops. Fixed!

**Comment by jarrpa** [comment]
@michaelgugino I updated this PR with the upgrade playbook as well, since that has additional refactorings that may help the situation. PTAL?

**Comment by jarrpa** [comment]
@michaelgugino CI tests passed.

**Comment by jarrpa** [comment]
@sdodson Added several patches of fixes found on master.

**Comment by sdodson** [comment]
> @sdodson Added several patches of fixes found on master.

Since CI doesn't validate CNS, has QE verified the fixes on master branch?

**Comment by jarrpa** [comment]
@sdodson CNS QE is waiting for this to merge on 3.10 before they can validate it.

**Pull request merged**